src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1946,13 +1925,21 @@ private ILabeledOperation CreateBoundLabeledStatementOperation(BoundLabeledState
 
         private IExpressionStatementOperation CreateBoundExpressionStatementOperation(BoundExpressionStatement boundExpressionStatement)
         {
-            IOperation expression = Create(boundExpressionStatement.Expression);
-            SyntaxNode syntax = boundExpressionStatement.Syntax;
-
             // lambda body can point to expression directly and binder can insert expression statement there. and end up statement pointing to
             // expression syntax node since there is no statement syntax node to point to. this will mark such one as implicit since it doesn't
             // actually exist in code
             bool isImplicit = boundExpressionStatement.WasCompilerGenerated || boundExpressionStatement.Syntax == boundExpressionStatement.Expression.Syntax;
+            SyntaxNode syntax = boundExpressionStatement.Syntax;
+
+            // If we're creating the tree for a speculatively-bound constructor initializer, there can be a bound sequence as the child node here
+            // that corresponds to the lifetime of any declared variables.
+            IOperation expression = Create(boundExpressionStatement.Expression);
+            if (boundExpressionStatement.Expression is BoundSequence sequence)
+            {
+                Debug.Assert(boundExpressionStatement.Syntax == sequence.Value.Syntax);
+                isImplicit = true;
```

Comment:
>isImplicit = true; [](start = 16, length = 18)

Can we observe effect of this adjustments in any test that asserts the shape of the node we create?

Reply:
I can update the verification in `Scope_ConstructorInitializers_06` to go two nodes up, currently it's just using the regular machinery.

---
In reply to: [527029812](https://github.com/dotnet/roslyn/pull/49453#discussion_r527029812) [](ancestors = 527029812)

Reply:
> I can update the verification in Scope_ConstructorInitializers_06 to go two nodes up, currently it's just using the regular machinery.

As we discussed offline, it would be good to assert the shape of these node since they can be navigated to.

---
In reply to: [527230521](https://github.com/dotnet/roslyn/pull/49453#discussion_r527230521) [](ancestors = 527230521,527029812)